lake/root.go

```diff
@@ -175,34 +175,35 @@ func (r *Root) readLakeMagic(ctx context.Context) error {
 	return nil
 }
 
-func (r *Root) batchifyPools(ctx context.Context, zctx *zed.Context, f expr.Filter) (zbuf.Array, error) {
+func (r *Root) batchifyPools(ctx context.Context, zctx *zed.Context, f expr.Filter) ([]zed.Value, error) {
 	m := zson.NewZNGMarshalerWithContext(zctx)
 	m.Decorate(zson.StylePackage)
 	pools, err := r.ListPools(ctx)
 	if err != nil {
 		return nil, err
 	}
-	var batch zbuf.Array
+	ectx := expr.NewContext()
+	var batch []zed.Value
```

Comment:
Nit: vals would be a better name for this variable in all these batchify* functions, but we can do that in another PR.